core/src/main/java/feast/core/model/Feature.java

```diff
@@ -32,19 +32,49 @@
 @Getter
 @Setter
 @Entity
-public class Feature extends Field {
+public class Feature {
+
+  @EmbeddedId private FeatureReference reference;
+
+  // Type of the field
+  private String type;
+
+  // Presence constraints (refer to proto feast.core.FeatureSet.FeatureSpec)
+  // Only one of them can be set.
+  private byte[] presence;
+  private byte[] groupPresence;
+
+  // Shape type (refer to proto feast.core.FeatureSet.FeatureSpec)
+  // Only one of them can be set.
+  private byte[] shape;
+  private byte[] valueCount;
+
+  // Domain info for the values (refer to proto feast.core.FeatureSet.FeatureSpec)
+  // Only one of them can be set.
+  private String domain;
+  private byte[] intDomain;
+  private byte[] floatDomain;
+  private byte[] stringDomain;
+  private byte[] boolDomain;
+  private byte[] structDomain;
+  private byte[] naturalLanguageDomain;
+  private byte[] imageDomain;
+  private byte[] midDomain;
+  private byte[] urlDomain;
+  private byte[] timeDomain;
+  private byte[] timeOfDayDomain;
 
   public Feature() {}
 
   private Feature(String name, ValueType.Enum type) {
-    this.setId(new FieldId());
-    this.setName(name);
+    this.setReference(new FeatureReference());
+    this.reference.setName(name);
     this.setType(type.toString());
   }
 
-  public static Feature withId(FieldId fieldId) {
+  public static Feature withReference(FeatureReference featureId) {
```

Comment:
is `with` used in practice? How about `fromRef`?

Reply:
`fromRef` doesn't convey the meaning though. It's called a `reference` but its more of a property of the class